sdk/src/deserialize_utils.rs

```diff
@@ -0,0 +1,91 @@
+use serde::{Deserialize, Deserializer};
+
+pub fn default_on_eof<'de, T, D>(d: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de> + Default,
+{
+    let result = T::deserialize(d);
+    match result {
+        Err(err) if err.to_string() == "io error: unexpected end of file" => Ok(T::default()),
```

Comment:
@mvines I spent a bit of time trying to check the error type here, but didn't make a lot of headway, due to the boxed generic error that serde deserialize returns (to accommodate different formats, etc). Got any ideas?

Reply:
I also spent time there and didn't make any progress, I think this is the best way

Reply:
ok, wfm